src/GraphMetadataViewExtension/GraphMetadataViewModel.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Windows.Media.Imaging;
+using Dynamo.Core;
+using Dynamo.Graph.Workspaces;
+using Dynamo.GraphMetadata.Controls;
+using Dynamo.UI.Commands;
+using Dynamo.Wpf.Extensions;
+
+namespace Dynamo.GraphMetadata
+{
+    public class GraphMetadataViewModel : NotificationObject
+    {
+        private readonly ViewLoadedParams viewLoadedParams;
+        private HomeWorkspaceModel currentWorkspace;
+
+        public DelegateCommand AddCustomPropertyCommand { get; set; }
+
+        public string GraphDescription
+        {
+            get { return currentWorkspace.Description; }
+            set { currentWorkspace.Description = value; RaisePropertyChanged(nameof(GraphDescription)); }
+        }
+
+        public string GraphAuthor
+        {
+            get { return currentWorkspace.Author; }
+            set { currentWorkspace.Author = value; RaisePropertyChanged(nameof(GraphAuthor)); }
+        }
+
+        public Uri HelpLink
+        {
+            get { return currentWorkspace.GraphDocumentationURL; }
+            set { currentWorkspace.GraphDocumentationURL = value; RaisePropertyChanged(nameof(HelpLink)); }
+        }
+
+        public BitmapImage Thumbnail
+        {
+            get
+            {
+                var bitmap = ImageFromBase64(currentWorkspace.Thumbnail);
+                return bitmap;
+            }
+            set
+            {
+                var base64 = value is null ? string.Empty : Base64FromImage(value);
+                currentWorkspace.Thumbnail = base64;
+                RaisePropertyChanged(nameof(Thumbnail));
+            }
+        }
+
+        public ObservableCollection<CustomPropertyControl> CustomProperties { get; set; }
+
+        public GraphMetadataViewModel(ViewLoadedParams viewLoadedParams)
+        {
+            this.viewLoadedParams = viewLoadedParams;
+            this.currentWorkspace = viewLoadedParams.CurrentWorkspaceModel as HomeWorkspaceModel;
+
+            this.viewLoadedParams.CurrentWorkspaceChanged += OnCurrentWorkspaceChanged;
+            // using this as CurrentWorkspaceChanged wont trigger if you:
+            // Close a saved workspace and open a New homeworkspace..
+            // This means that properties defined in the previous opened workspace will still be showed in the extension.
+            // CurrentWorkspaceCleared will trigger everytime a graph is closed which allows us to reset the properties. 
+            this.viewLoadedParams.CurrentWorkspaceCleared += OnCurrentWorkspaceChanged;
+
+            CustomProperties = new ObservableCollection<CustomPropertyControl>();
+            InitializeCommands();
+        }
+
+        private void OnCurrentWorkspaceChanged(Graph.Workspaces.IWorkspaceModel obj)
+        {
+            if (!(obj is HomeWorkspaceModel hwm)) return;
```

Comment:
Should we clear properties in this case as well ? How this happen actually ?

Reply:
Changed this a bit, so now if the workspace is changed to something that isnt a `HomeWorkspaceModel` (i believe the only other scenario currently is `CustomNodeWorkspace`) it will close the extension. I dont think we would want to clear the custom properties in this scenario as users might just want to jump into a custom node and then back to the main graph, in that case i assuem they would still want all of the Custom props they created.

However currently all of the custom properties are being reset in this scenario, this is because when switching from Custom Node ws to Home ws will trigger workspace changed which clears all custom properties. 

Lets discuss later today what we expect to happen in that scenario

Reply:
Ok